### PR TITLE
Update CHANGELOG for release 1.3

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,36 @@
...
+- Require C++17 [\#715](https://github.com/arborx/ArborX/pull/715)
+- Require Kokkos 3.6 [\#748](https://github.com/arborx/ArborX/pull/748)
+
+**Enhancements:**
+- Switch to using stackless spatial traversal for all backends [\#672](https://github.com/arborx/ArborX/pull/672)
+- Add a ray-tracing example using ordered tree traversal [\#691](https://github.com/arborx/ArborX/pull/691)
+- Add `distance()` function to the nearest predicate [\#700](https://github.com/arborx/ArborX/pull/700)
+- Template `BruteForce` on bounding volume [\#712](https://github.com/arborx/ArborX/pull/712)
+- Warn about a potential loss of precision when using FDBSCAN-DenseBox [\#560](https://github.com/arborx/ArborX/pull/560)
+- Migrate DBSCAN driver from `examples/` to `benchmarks/` [\#736](https://github.com/arborx/ArborX/pull/736)
```
Comment (dalg24) [inline CHANGELOG.md:21]: Not an enhancement and we should probably add a proper example before we release

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+**Build changes:**
+- Require C++17 [\#715](https://github.com/arborx/ArborX/pull/715)
+- Require Kokkos 3.6 [\#748](https://github.com/arborx/ArborX/pull/748)
+
+**Enhancements:**
+- Switch to using stackless spatial traversal for all backends [\#672](https://github.com/arborx/ArborX/pull/672)
+- Add a ray-tracing example using ordered tree traversal [\#691](https://github.com/arborx/ArborX/pull/691)
+- Add `distance()` function to the nearest predicate [\#700](https://github.com/arborx/ArborX/pull/700)
+- Template `BruteForce` on bounding volume [\#712](https://github.com/arborx/ArborX/pull/712)
+- Warn about a potential loss of precision when using FDBSCAN-DenseBox [\#560](https://github.com/arborx/ArborX/pull/560)
```
Comment (dalg24) [inline CHANGELOG.md:20]: It is more than a warning, we throw an exception

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+- Migrate DBSCAN driver from `examples/` to `benchmarks/` [\#736](https://github.com/arborx/ArborX/pull/736)
+
+**Deprecations:**
+- Throw an error when using the old `ArborX::Traits` interface [\#667](https://github.com/arborx/ArborX/pull/667)
```
Comment (dalg24) [inline CHANGELOG.md:24]: We do not throw we make compilation fail via a static_assert

**Review by dalg24 — CHANGES_REQUESTED** [review]

**Review by dalg24 — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,35 @@
...
+
+**Enhancements:**
+- Switch to using stackless spatial traversal for all backends [\#672](https://github.com/arborx/ArborX/pull/672)
+- Add ordered tree traversal (EXPERIMENTAL) [\#683](https://github.com/arborx/ArborX/pull/683)
+- Add a ray-tracing example using ordered tree traversal [\#691](https://github.com/arborx/ArborX/pull/691)
+- Add `distance()` function to the nearest predicate [\#700](https://github.com/arborx/ArborX/pull/700)
+- Template `BruteForce` on bounding volume [\#712](https://github.com/arborx/ArborX/pull/712)
+- Throw an exception when FDBSCAN-DenseBox may potentially produce wrong result [\#560](https://github.com/arborx/ArborX/pull/560)
+- Migrate full DBSCAN driver from `examples/` to `benchmarks/` [\#736](https://github.com/arborx/ArborX/pull/736)
+- Add a DBSCAN example [\#763](https://github.com/arborx/ArborX/pull/763)
```
Comment (dalg24) [inline CHANGELOG.md:22]: I would have merged the two

Comment (aprokop) [reply CHANGELOG.md:22]: I like them separately better.

**Pull request merged**